lib/tasks/ensure-v2-addons.js

```diff
@@ -63,25 +60,35 @@ export default async function ensureV2Addons() {
   }
 }
 
-async function getV1Addons() {
+async function getV1Addons({ errorTrace = false } = {}) {
   const packageJSON = JSON.parse(await readFile('package.json', 'utf8'));
   const deps = { ...packageJSON.devDependencies, ...packageJSON.dependencies };
   const v1packages = [];
-  for (const depName in deps) {
+  for (const [depName, version] of Object.entries(deps)) {
     try {
-      const pkg = JSON.parse(
-        await readFile(join('node_modules', depName, 'package.json'), 'utf8'),
-      );
+      const { stdout } =
+        await execa`npm view --json ${depName}@${version} ember-addon keywords`;
+      const pkg = JSON.parse(stdout);
       if (
         pkg.keywords?.includes('ember-addon') &&
         pkg['ember-addon']?.version !== 2
       ) {
         v1packages.push(depName);
       }
     } catch (e) {
-      console.warn(
-        `The package ${depName} was found in the dependencies, but its package.json was not found. Is the package correctly installed? ${e}`,
-      );
+      if (e.message.includes('npm error code E404')) {
+        console.warn(
+          `We could not look up information about ${depName}. You need to verify if this addon is a v2 addon manually.`,
+        );
+      } else {
+        console.warn(
+          `We recieved an unexpected error looking up information about ${depName}.`,
+        );
```

Comment:
A tiny suggestion: I would not use "we" in the error message since the user doesn't really know who that is.

```suggestion
        console.warn(
          `Could not look up information about ${depName}. You need to verify if this addon is a v2 addon manually.`,
        );
      } else {
        console.warn(
          `Unexpected error received looking up information about ${depName}.`,
        );
```